langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Slice.java

```diff
@@ -70,18 +73,24 @@ public static ArrayValue slice(Strand strand, ArrayValue arr, long startIndex, l
 
         BType arrType = arr.getType();
         ArrayValue slicedArr;
-        int elemTypeTag;
 
         switch (arrType.getTag()) {
             case TypeTags.ARRAY_TAG:
                 slicedArr = ((ArrayValueImpl) arr).slice(startIndex, endIndex);
                 break;
             case TypeTags.TUPLE_TAG:
                 BTupleType tupleType = (BTupleType) arrType;
-                BUnionType unionType = new BUnionType(tupleType.getTupleTypes(), tupleType.getTypeFlags());
+
+                List<BType> memTypes = new ArrayList<>(tupleType.getTupleTypes());
+
+                BType restType = tupleType.getRestType();
+                if (restType != null) {
+                    memTypes.add(restType);
+                }
+
+                BUnionType unionType = new BUnionType(memTypes);
```

Comment:
Shouldn't the new type have the same flags as the original type?

Reply:
Not really. For the array type maybe we could use the same but not for the union type.

The flags are `NILABLE`, `ANYDATA`, `PURE`.

Consider the tuple type `[int, string?, error...]`. The is `PURE` and `ANYDATA`.

The union created for this is  `int|string|error?`. This is `NILABLE` and `PURE`.

This `BUnionType` constructor computes the new flags - https://github.com/ballerina-platform/ballerina-lang/blob/ballerina-1.2.x/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BUnionType.java#L56.